### PR TITLE
OPAL-694

### PR DESCRIPTION
**Pull request by JS-USAF**

### CODE REVIEW

File: weave/index/index_pandas.py [hunk 1]
```diff
@@ -134,8 +138,11 @@ def sync_index(self):
             self.generate_index()
             return
         if len(index_paths) > 20:
-            warnings.warn(f"The index basket count is {len(index_paths)}. " +
-                 "Consider running weave.IndexPandas.clean_up_indices")
+            if self.auto_cleanup:
+                self.clean_up_indices()
+            else:
```
Comment (JN-USAF) [inline weave/index/index_pandas.py:143]: I think there's an issue with the text formatting here. When I test it out myself, I see the following: some extra text:
image

File: weave/index/index_pandas.py [hunk 2]
```diff
@@ -134,8 +138,11 @@ def sync_index(self):
             self.generate_index()
             return
         if len(index_paths) > 20:
-            warnings.warn(f"The index basket count is {len(index_paths)}. " +
-                 "Consider running weave.IndexPandas.clean_up_indices")
+            if self.auto_cleanup:
```
Comment (JN-USAF) [inline weave/index/index_pandas.py:141]: I'm assuming that auto_cleanup statement should be the default, but when I ran it for the first time with more than 20 buckets added, it skipped this statement and went straight to the user warning. Is that expected?

Comment (JN-USAF) [reply weave/index/index_pandas.py:141]: My mistake, it does work as expected. I was on the wrong branch

File: weave/index/index_pandas.py (hunk 1 above)
Comment (JS-USAF) [reply weave/index/index_pandas.py:143]: I updated this so that the warning.warn does not look cut off by using a variable warn instead of the full string

**Review by CW-USAF — APPROVED** [review]

**Review by JN-USAF — APPROVED** [review]

**Pull request merged**